config/routes/sidekiq_routes.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module SidekiqRoutes
+  def self.extended(router)
+    router.instance_exec do
+      # Sidekiq Basic Auth implementation
+      # https://github.com/mperham/sidekiq/wiki/Monitoring#rails-http-basic-auth-from-routes
+      require "sidekiq/web"
+
+      if Rails.env.production?
+        Sidekiq::Web.use Rack::Auth::Basic do |username, password|
+          sidekiq_username = Rails.application.credentials.dig(Rails.env.to_sym, :sidekiq, :username)
+          sidekiq_password = Rails.application.credentials.dig(Rails.env.to_sym, :sidekiq, :password)
+          utils = ActiveSupport::SecurityUtils
+
+          utils.secure_compare(::Digest::SHA256.hexdigest(sidekiq_username), ::Digest::SHA256.hexdigest(username)) &
+            utils.secure_compare(::Digest::SHA256.hexdigest(sidekiq_password), ::Digest::SHA256.hexdigest(password))
```

Comment:
Will it be worth integrating with the app's auth once we have it?

Reply:
as discussed in tech breakdown this will be dealt separately